actions/plugins/version-bump-changelog/action.yml

```
@@ -0,0 +1,113 @@
+
+    - name: Generate GitHub token
+      id: generate-github-token
+      uses: actions/create-github-app-token@3ff1caaa28b64c9cc276ce0a02e2ff584f3900c5 # v2.0.2
+      with:
+        app-id: ${{ env.GITHUB_APP_ID }}
+        private-key: ${{ env.GITHUB_APP_PRIVATE_KEY }}
+        owner: ${{ github.repository_owner }}
+
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      with:
+        token: ${{ steps.generate-github-token.outputs.token }}
+        persist-credentials: true
+        fetch-depth: 0
+
+    - name: Setup Git
+      shell: bash
+      run: |
+        git config user.name 'grafana-plugins-platform-bot[bot]'
+        git config user.email '144369747+grafana-plugins-platform-bot[bot]@users.noreply.github.com'
+
+    - name: Get previous tag before bumping
+      id: previous-tag
+      if: ${{ inputs.generate-changelog == 'true' }}
+      shell: bash
+      run: |
+        PREVIOUS_TAG=$(git describe --tags --abbrev=0 2>/dev/null || echo "v0.0.0")
+        echo "previous-tag=${PREVIOUS_TAG}" >> $GITHUB_OUTPUT
+
+    - name: Bump version
+      id: bump
+      shell: bash
+      run: |
+        NEW_VERSION=$(npm version ${INPUT_VERSION} --no-git-tag-version)
+        echo "new-version=${NEW_VERSION}" >> $GITHUB_OUTPUT
+      env:
+        INPUT_VERSION: ${{ inputs.version }}
+        GITHUB_TOKEN: ${{ steps.generate-github-token.outputs.token }}
+    
+    - name: Generate changelog
+      if: ${{ inputs.generate-changelog == 'true' }}
+      shell: bash
+      run: |
+        # Generate changelog
+        if [[ "${{ steps.previous-tag.outputs.previous-tag }}" == "v0.0.0" ]]; then
+          echo "No previous tag found ${{ steps.previous-tag.outputs.previous-tag }}, generating changelog for initial version only: v${{ steps.bump.outputs.new-version }}"
+          # For initial version, get all commits up to HEAD
+          npx generate-changelog -t HEAD
+        else
+          echo "Generating changelog from ${{ steps.previous-tag.outputs.previous-tag }} to HEAD"
+          npx generate-changelog -t ${{ steps.previous-tag.outputs.previous-tag }}...HEAD
+        fi
+
+        # Remove existing "Changelog" header anywhere in the file
+        sed -i '/^#\sChangelog$/Id' CHANGELOG.md
+        # Prepend the header to the top of the file
+        printf "# Changelog\n\n" | cat - CHANGELOG.md > temp && mv temp CHANGELOG.md
+        # Format headers remove dates and unreleased
+        sed -i -E 's/^(#+ [0-9]+\.[0-9]+\.[0-9]+) \(([Uu]nreleased|[0-9]{4}-[0-9]{2}-[0-9]{2})\)/\1/' CHANGELOG.md
+
+      env:
+        GITHUB_TOKEN: ${{ steps.generate-github-token.outputs.token }}
+
+    - name: Commit changes
+      shell: bash
+      run: |
+        git add package.json
+        git add CHANGELOG.md || true  # No-op if changelog not generated
+        git commit -m "chore(version): bump version to ${{ steps.bump.outputs.new-version }}"
+        git push origin main
```

```suggestion
        git tag -a ${{ steps.bump.outputs.new-version }} -m "Release version ${{ steps.bump.outputs.new-version }}"
        git push origin main --tags
```

you can do the next two steps right here